### PR TITLE
ads: Do not log Envoy's XDS Certificate CN; log cert's SerialNumber instead

### PR DESCRIPTION
**Pull request by draychev**
This PR removes XDS Certificate CommonName from log messages and uses XDS Certificate's SerialNumber.

---

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

File: pkg/catalog/proxy.go [hunk 1]
```diff
@@ -26,9 +26,14 @@ func (mc *MeshCatalog) RegisterProxy(proxy *envoy.Proxy) {
...
 		podUID := types.UID(proxy.PodMetadata.UID)
+
+		// Create a PodUID to Certificate CN map so we can easily determine the CN from the PodUID
 		mc.podUIDToCN.Store(podUID, proxy.GetCertificateCommonName())
+
+		// Create a PodUID to Cert Serial Number so we can easily look-up the SerialNumber of the cert issued to a proxy for a given Pod.
+		mc.podUIDToCertificateSerialNumber.Store(podUID, proxy.GetCertificateSerialNumber())
 	}
-	log.Info().Msgf("Registered new proxy: CN=%v, ip=%v", proxy.GetCertificateCommonName(), proxy.GetIP())
+	log.Trace().Msgf("Registered new proxy with certificate SerialNumber=%s for Pod with UID=%s", proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
```
Comment (snehachhabria) [inline pkg/catalog/proxy.go:36]:
```suggestion
	log.Debug().Msgf("Registered new proxy with certificate SerialNumber=%s for Pod with UID=%s", proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
```

File: pkg/catalog/proxy.go [hunk 2]
```diff
@@ -39,5 +44,5 @@ func (mc *MeshCatalog) UnregisterProxy(p *envoy.Proxy) {
 		lastSeen: time.Now(),
 	})
 
-	log.Info().Msgf("Unregistered proxy: CN=%v, ip=%v", p.GetCertificateCommonName(), p.GetIP())
+	log.Trace().Msgf("Unregistered proxy with certificate SerialNumber=%v for Pod with UID=%s", p.GetCertificateSerialNumber(), p.GetPodUID())
```
Comment (snehachhabria) [inline pkg/catalog/proxy.go:47]:
```suggestion
	log.Debug().Msgf("Unregistered proxy with certificate SerialNumber=%v for Pod with UID=%s", p.GetCertificateSerialNumber(), p.GetPodUID())
```

**Review by snehachhabria — APPROVED** [review]

File: pkg/envoy/cds/response.go [hunk 3]
```diff
@@ -78,13 +78,13 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 	alreadyAdded := mapset.NewSet()
 	for _, cluster := range clusters {
 		if alreadyAdded.Contains(cluster.Name) {
-			log.Error().Msgf("Found duplicate clusters with name %s; Duplicate will not be sent to Envoy for Service %s with CN=%s", cluster.Name, proxyServiceName, proxy.GetCertificateCommonName())
+			log.Error().Msgf("Found duplicate clusters with name %s; Duplicate will not be sent to Envoy with XDS Certificate SerialNumber=%s on on Pod with UID=%s", cluster.Name, proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
```
Comment (SanyaKochhar) [inline pkg/envoy/cds/response.go:81]: nit: "on on"

File: pkg/envoy/cds/response.go [hunk 4]
```diff
@@ -78,13 +78,13 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
...
 			continue
 		}
 		alreadyAdded.Add(cluster.Name)
 		marshalledClusters, err := ptypes.MarshalAny(cluster)
 		if err != nil {
-			log.Error().Err(err).Msgf("Failed to marshal cluster %s for proxy %s", cluster.Name, proxy.GetCertificateCommonName())
+			log.Error().Err(err).Msgf("Failed to marshal cluster %s for Envoy with XDS Certificate SerialNumber=%s on on Pod with UID=%s", cluster.Name, proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
```
Comment (SanyaKochhar) [inline pkg/envoy/cds/response.go:87]: nit: same as above

Comment (draychev) [reply pkg/envoy/cds/response.go:87]: fixed!

File: pkg/envoy/cds/response.go (hunk 3 above)
Comment (draychev) [reply pkg/envoy/cds/response.go:81]: fixed!

**Review by SanyaKochhar — APPROVED** [review]

**Pull request merged**